### PR TITLE
Update developer guide to include list

### PR DESCRIPTION
**Pull request by zoranabc201**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -365,6 +392,49 @@ The sequence diagram of `updateClient`, `updateProperty` and `updatePair` can be
...
+* There are five different classes which handle each of the features described above. Each class inherits from the
+   abstract Command class, and reads information present in either the PropertyList or ClientList objects respectively.
+
+The execute function retrieves the propertyList holding all the current properties. It then loops 
+through every property present. For every property, it passes it to the corresponding display function
+in Ui.
+
+* Each of these commands then uses a method present in the Ui class, to print an individual client, or property.
+   The loop for printing every single client or property is present in the Command itself.
+  ![CommandListClientsClass](diagrams/CommandListPropertiesClass.png)
```
Comment (FeliciaBeatrice) [inline docs/DeveloperGuide.md:434]: small typo for displayOneProperty

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -68,6 +68,33 @@ ___
 ### Parser Component
 ### Client Component
 ### Property Component
+### List Component
+The list feature has the following commands in it -
```
Comment (FeliciaBeatrice) [inline docs/DeveloperGuide.md:72]: good explanation of each command

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -365,6 +392,49 @@ The sequence diagram of `updateClient`, `updateProperty` and `updatePair` can be
...
+### List feature
+
+There are 3 main steps whenever a list command needs to be executed
+* When the user enters any command, it first needs be understood. That is handled by the ParseManager class.
+   Next, when the first word entered by the user is determined to be `list`, ParseManager itself then determines
+   the type of list command entered, including the tags. ParseListClient, ParseListProperty and ParseListEverything
+   are checkers to ensure that a valid command has been entered. ParseListClient and ParseListProperty also determine
+   if tags have been entered, and if those tags are valid.
+  ```
+        case COMMAND_LIST:
```
Comment (FeliciaBeatrice) [inline docs/DeveloperGuide.md:405]: maybe can summarize the code so that not a whole block of code is shown (?)

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -68,6 +68,33 @@ ___
...
+* `list -property FLAG` This lists only the information present in the TAG for every property. The types
+  of TAG are -
+    * `a/` This is for address
+    * `n/` This is for name
+    * `p/` This is for price
+    * `t/` This is for type  
+
+There are 5 different classes, that each inherit from the abstract Command class. The commands read information from 
+the PropertyList and ClientList classes respectively, and display using the Ui class, making use of the objects of 
+these classes. The Commands which display all the information - i.e. CommandListClients, CommandListProperties, and
```
Comment (wilsonngja) [inline docs/DeveloperGuide.md:91]: Can consider breaking up or rephrasing this sentence.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -365,6 +392,49 @@ The sequence diagram of `updateClient`, `updateProperty` and `updatePair` can be
...
+-property or -everything) and returns the corresponding object.  
+Both ParseListClient and ParseListObject then determine if tags are present, and if they are valid, 
+throwing exceptions if any errors are encountered. They then return the corresponding Command type necessary.
+
+* There are five different classes which handle each of the features described above. Each class inherits from the
+   abstract Command class, and reads information present in either the PropertyList or ClientList objects respectively.
+
+The execute function retrieves the propertyList holding all the current properties. It then loops 
+through every property present. For every property, it passes it to the corresponding display function
+in Ui.
```
Comment (wilsonngja) [inline docs/DeveloperGuide.md:430]: Can consider rephrasing this paragraph as well. The first sentence is not very coherent. For the last sentence, the part on "it passes it" can be rephrase.

**Review by wilsonngja — APPROVED** [review]
Overall, good effort for the write up for Developer Guide. Can consider improving on the language used to help to improve the coherence in v2.1.

**Pull request merged**